staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -1079,3 +1109,99 @@ func notFound(key string) clientv3.Cmp {
 func getTypeName(obj interface{}) string {
 	return reflect.TypeOf(obj).String()
 }
+
+// checkVersionEtcdClientInterface defines the interface required for testing purposes in IsDeprecated function of the interface.
```

Comment:
Can you please move all of that to a separate file - this is relatively well separate logic and growing the files in this whole package indefinitely makes it harder to follow.